app/Http/Controllers/CategoryController.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\CategoryRequest;
+use App\Models\Category;
+use Illuminate\Http\Request;
+
+class CategoryControler extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $categories = Category::with('parent', 'childCategories')
+            ->get();
+
+        return view('admin.categories.list', compact('categories'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $categories = Category::where('parent_id', '=', config('uploads.category_root'))
+            ->with('childCategories')
+            ->get();
+
+        return view('admin.categories.add', compact('categories'));
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(CategoryRequest $request)
+    {
+        $category = new Category();
+        $parentId = ($request->category != '' ? $request->category : config('uploads.category_root'));
```

Comment:
```suggestion
        $parentId = ($request->category ?? config('uploads.category_root'));
```
đọc thêm về ?? xem có đáp ứng được yêu cầu của e ko nhé

Reply:
dạ được ạ ~